src/packagedef.jl

```diff
@@ -1345,6 +1345,13 @@ function __init__()
         return nothing
     end
 
+    # Setting up the paths relative to package module location
+
+    global juliadir = find_juliadir()
+    global basesrccache = normpath(joinpath(expected_juliadir(), "base.cache"))
+    global depsdir = joinpath(isnothing(pkgdir(@__MODULE__)) ? dirname(@__DIR__) : pkgdir(@__MODULE__), "deps")
+    silencefile[] = joinpath(depsdir, "silence.txt")
```

Comment:
Tangentially, the whole `silence` functionality should be rewritten using Preferences.jl instead of mutating the package folder.

Reply:
I am currently overwhelmed and refactoring to use Preferences.jl is not on my plate, but I support the change.